Golangci-linter fails "test"

Did we merge two PRs without rebasing correctly again? The main is now broken.

> > @tuminoid: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> > Test name	Commit	Details	Required	Rerun command
> > test	[b0c82c2](https://github.com/metal3-io/baremetal-operator/commit/b0c82c20b36aeea19e4ca974e376a931aa9c52ad)	[link](https://prow.apps.test.metal3.io/view/s3/prow-logs/pr-logs/pull/metal3-io_baremetal-operator/3092/test/2034175865402691584)	unknown	`/test test`
> > [Full PR test history](https://prow.apps.test.metal3.io/pr-history?org=metal3-io&repo=baremetal-operator&pr=3092). [Your PR dashboard](https://prow.apps.test.metal3.io/pr?query=is:pr+state:open+author:tuminoid).
> > Details
> 
> Sigh, completely unrelated golangci-lint failures....

```
pkg/secretutils/label.go:1: : # github.com/metal3-io/baremetal-operator/pkg/secretutils [github.com/metal3-io/baremetal-operator/pkg/secretutils.test]
pkg/secretutils/secret_manager_test.go:268:66: too many arguments in call to NewSecretManager
	have (context.Context, logr.Logger, client.WithWatch, client.WithWatch)
	want (logr.Logger, client.Client, client.Reader)
pkg/secretutils/secret_manager_test.go:270:103: not enough arguments in call to sm.AcquireSecret
	have (types.NamespacedName, *"github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1".BareMetalHost, bool)
	want (context.Context, types.NamespacedName, client.Object, bool)
pkg/secretutils/secret_manager_test.go:324:66: too many arguments in call to NewSecretManager
	have (context.Context, logr.Logger, client.WithWatch, client.WithWatch)
	want (logr.Logger, client.Client, client.Reader)
pkg/secretutils/secret_manager_test.go:326:103: not enough arguments in call to sm.AcquireSecret
	have (types.NamespacedName, *"github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1".BareMetalHost, bool)
	want (context.Context, types.NamespacedName, client.Object, bool) (typecheck)
package secretutils
1 issues:
* typecheck: 1
level=info msg="File cache stats: 1 entries of total size 895B"
level=info msg="Memory: 386 samples, avg is 429.2MB, max is 1921.0MB"
level=info msg="Execution took 38.473495405s"
make: *** [Makefile:170: lint] Error 1
```

_Originally posted by @tuminoid in https://github.com/metal3-io/baremetal-operator/issues/3092#issuecomment-4080455859_

--- a/pkg/secretutils/secret_manager_test.go
+++ b/pkg/secretutils/secret_manager_test.go
@@ -265,9 +265,9 @@ func TestSecretManager_AcquireSecret_DeletesController(t *testing.T) {
 
 	fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(secret, owner).Build()
 
-	sm := NewSecretManager(t.Context(), logr.Discard(), fakeClient, fakeClient)
+	sm := NewSecretManager(logr.Discard(), fakeClient, fakeClient)
 
-	result, err := sm.AcquireSecret(types.NamespacedName{Name: "test-secret", Namespace: "test"}, owner, false)
+	result, err := sm.AcquireSecret(t.Context(), types.NamespacedName{Name: "test-secret", Namespace: "test"}, owner, false)
 	require.NoError(t, err)
 	assert.Equal(t, "test-secret", result.Name)
 
@@ -321,9 +321,9 @@ func TestSecretManager_AcquireSecret_KeepsOtherOwners(t *testing.T) {
 
 	fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(secret, owner).Build()
 
-	sm := NewSecretManager(t.Context(), logr.Discard(), fakeClient, fakeClient)
+	sm := NewSecretManager(logr.Discard(), fakeClient, fakeClient)
 
-	result, err := sm.AcquireSecret(types.NamespacedName{Name: "test-secret", Namespace: "test"}, owner, false)
+	result, err := sm.AcquireSecret(t.Context(), types.NamespacedName{Name: "test-secret", Namespace: "test"}, owner, false)
 	require.NoError(t, err)
 	assert.Equal(t, "test-secret", result.Name)
 

